atlp-rwanda/Team-furebo-E-commerce-bn
# ft-change-password

#### What does this PR do?
      * User should be able to chang or update her/his password

#### Description of Task to be completed?
      * A User has to login first.
    - Check if user logged in first
    - Check if currentPassword is correct
    - Check if new password is not same as currentPassword
    - Avoid user to change someone else's password
    - User has to have account first to be able to change password
    - ONCE ALL THAT CHECKED OLD PASSWORD BE REPLACED with new one

#### How should this be manually tested?
  * Clone the branch
  * Go to its path in terminal run: npm install for Dependencies
  * Create .env file
  * Add the following Variables in that .env file
    - JWT_SECRET
    - USER_SECRET_KEY
    - BCRYPT_KEY
    - TOKEN_EXPIRATION
    - SENDGRID_API_KEY
    - SENDER_EMAIL
    - GOOGLE_CLIENT_ID
    - GOOGLE_CLIENT_SECRET
    - NODE_ENV
    - DB_USERNAME
    - DB_PASSWORD
    - DB_HOST
    - DB_NAME
    - RESET_URL

  * Go to this URL to test functionalities: http://localhost:5000/api-docs/
  * Go to user Conner 
  * Make sure to Create Account
      * CHANGE PASSWORD

#### What are the relevant pivotal tracker/Trello stories?
      * N/A
